Assets/Text/JavaScript/NodeScripts/VelocityDampener.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Dampens the velocity components of a BodyNode.
+ * 
+ * Script written by Aiden Ravenshea.
+ */
+class VelocityDampener {
+
+    constructor(node) {
+        this.node = node;
+        this.linearForce = 5; // Linear velocity dampening force
+        this.angularForce = 5; // Angular velocity dampening force
+        this.scene = this.node.findSceneNode();
+    }
+
+    update(delta) {
+        // Get dampening forces and dampen velocity components
+        const fv = Math.min(1, this.linearForce * this.scene.timeStep);
+        this.node.linearVelocityX -= this.node.linearVelocityX * fv;
+        this.node.linearVelocityY -= this.node.linearVelocityY * fv;
+        this.node.angularVelocity -= this.node.angularVelocity * Math.min(1, this.angularForce * this.scene.timeStep);
```

Comment:
Could we assign this Math,min at the top as well (for consistency)

Reply:
File has been updated. Thanks!